operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -416,17 +422,25 @@ func extractS3ConfigSecret(s *corev1.Secret, credentialMode lokiv1.CredentialMod
 		if len(roleArn) != 0 {
 			return nil, fmt.Errorf("%w: %s", errSecretFieldNotAllowed, storage.KeyAWSRoleArn)
 		}
+
+		err := validateS3Endpoint(endpoint, region)
```

Comment:
Why not pull the `validateS3Endpoint` out of the switch case? Then if the endpoint is != nil we want to run the validation. For instance, in STS I don't think that endpoint is a required field.